engine/runtime-parser/src/main/scala/org/enso/compiler/core/Implicits.scala

```diff
@@ -91,7 +88,7 @@ object Implicits {
       * @return the metadata for `pass`, if it exists
       */
     def getMetadata[K <: ProcessingPass](pass: K): Option[pass.Metadata] = {
-      ir.passData.get(pass)
+      ir.passData.get(pass).asInstanceOf[Option[pass.Metadata]]
```

Comment:
The type-system of Java doesn't support "abstract types". Doing something like that in Java would require a bit of overhaul - easier to just cast at few places.